app/frontend/src/features/userQueries/useUsers.ts

```diff
@@ -58,7 +64,7 @@ export default function useUsers(ids: number[], invalidate: boolean = false) {
   };
 }
 
-export function useUser(id: number, invalidate: boolean = false) {
+export function useUser(id: number | undefined, invalidate: boolean = false) {
```

Comment:
When do we start getting `undefined` user IDs? Seems a bit redundant to store an unknown user that doesn't exist?

Reply:
Yeah it's weird but the reason is control messages can have a "target" user (like if I remove you as an admin, you are the target) so I needed `useUser` for the target, but only some types of control messages have that so it might be undefined. And you can't have conditional hooks.

Reply:
Oh I see. Couldn't we filter that out in `useUsers` in the `ids.map` bit instead?

Reply:
It would be a little convoluted. Because `ControlMessageView` uses `useUser` not `useUsers`, so you'd have to put the target user id in a variable, put it in a list with author id, then use `useUsers`, filtering out undefined, and use the result map on id with the saved ids. I think just accepting undefined is more concise..

Reply:
Since `useUser` (singular) just do `useUsers([id])`, wouldn't the filter just work if you do it in `useUsers`'s `ids`?I was thinking literally `ids.filter(Boolean).map(...)` in `useQueries` would do the trick.

Then the rest would just return back together with an empty map if `id` of `useUser` turns out to be `undefined`, which shouldn't cause that much problem?

Reply:
Do you mean like this in 7f5a2f2ed75060f9c7c040833c2813845b28decc? Does it make a difference?

Reply:
Yeah that would remove nulls, undefined and 0 before they get mapped to queries. Probably fine to remove 0 since ids always start at 1 right from the backend?